leshan-core/src/main/java/org/eclipse/leshan/core/attr/Attachment.java

```diff
@@ -0,0 +1,10 @@
+package org.eclipse.leshan.core.attr;
+
+/**
+ * The attachment level of an LwM2m attribute.
+ */
+public enum Attachment {
```

Comment:
I think it will be necessary to be able to compile a proper DiscoverResponse, e.g. if you perform Discover on a resource, the inherited attributes from Object and Instance level should be included only if they have Attachment = Resource